### PR TITLE
Remove unsafe code

### PR DESCRIPTION
**Pull request by stepancheg**

### CODE REVIEW

File: starlark/src/syntax/lexer.rs [hunk 1]
```diff
@@ -350,18 +346,23 @@ impl Lexer {
     }
 
     fn peek(&mut self) -> Option<(u64, char)> {
-        match self.chars.peek() {
-            Some(&(x, c)) => Some((x as u64 + self.offset, c)),
+        match self.input[self.pos..].chars().next() {
```
Comment (damienmg) [inline starlark/src/syntax/lexer.rs]: This is a O(self.input.size) operation

File: starlark/src/syntax/lexer.rs [hunk 2]
```diff
@@ -350,18 +346,23 @@ impl Lexer {
...
+            Some(c) => Some((self.pos as u64 + self.offset, c)),
             None => None,
         }
     }
 
     fn pop(&mut self) -> Option<(u64, char)> {
-        let r = self.chars.next();
-        self.last_next = match r {
-            Some((x, c)) => {
+        let mut char_indices = self.input[self.pos..].char_indices();
```
Comment (damienmg) [inline starlark/src/syntax/lexer.rs]: Here too.

**Comment by stepancheg** [comment]
No, string indexing operation is `O(1)`, because it accepts byte index, not char index.

image

**Comment by stepancheg** [comment]
Renamed `pos` to `pos_bytes` and added a short comment.

**Pull request merged**